docs/coding-conventions.md

```diff
@@ -0,0 +1,297 @@
+about conventions in detail take a look at the [detailed
+coding conventions](#detailed-coding-conventions).
+
+- [Write elegant, cohesive and easily readable code](#write-elegant-cohesive-and-easily-readable-code)
+- [Avoid nesting and complexity by using early returns](#avoid-nesting-and-complexity-by-using-early-returns)
+- [Use the same coding style throughout a file](#use-the-same-coding-style-throughout-a-file)
+- [Avoid repetition of hardcoded values](#avoid-repetition-of-hardcoded-values)
+- [Uniform import order and naming](#uniform-import-order-and-naming)
+- Prefert to inline variable assignments.
+    - For example: Inline `err` checks in if-statements.
+- Use switch-cases to avoid long if/else if/else statements.
+- Isolate code.
+    - Use interfaces instead of bare structs.
+- Avoid use of global variables.
+    - Use structs and receiver methods to keep state.
+- Avoid long files.
+    - Avoid adding helpers in a single place. Long files like `tests/utils.go`
+      are hard to maintain.
+    - Alternatively, add helpers to the places where they are used or group them
+      in packages/files with representative names.
+- Prefer to define variables in the function body
+    - Avoid returning too many values from a function.
+    - Avoid naked returns.
+- Use closures with caution, be aware of the risks and use them only when it
+  makes sense.
+    - They can still be used if there is a good reason for it.
+    - A useful use case of closures is to define a function that is relevant
+      only for a very specific scope.
+- Declare empty slices with the var syntax.
+    - Pay attention when serializing data, see [Declaring Empty Slices](https://go.dev/wiki/CodeReviewComments#declaring-empty-slices).
+- Avoid use of `fmt.Sprintf` for manual construction of complex objects
+  or operations (e.g. paths or patches).
+    - Use helpers or builders when available.
+        - Build patches with the `PatchSet` interface.
+        - Construct paths with the `path` package.
+- Use the `kubevirt.io/kubevirt/pkg/pointer` package when pointers are
+  required.
+- Keep function signatures lean.
+    - E.g. use `kubevirt.Client()` in test functions instead of passing the
+      client as an additional function call argument.
+    - Again: Use structs and receiver methods to keep state.
+- Table-driven tests are preferred for testing matrices of scenarios/inputs.
+    - Use Gingko's `DescribeTable` to construct test tables.
+- Do not expect an asynchronous thing to happen immediately.
+    - For example do not wait for one second and expect a VM to be running.
+    - Wait and retry instead, in test code use `Eventually`.
+- Avoid `Skip` in tests.
+    - Use `decorators` to control in which lanes tests run.
+- Namings (packages, interfaces, etc.)
+    - Consider the package name when selecting an interface name and avoid
+      redundancy.
+      - For example, `storage.Interface` is better than `storage.
+        StorageInterface`.
```

Comment:
Well, if you need to use the type name (`interface`) in the name, you surely are doing something very wrong. An interface describes a behavior of an object, not an object.

Bad: `storage.Interface`.
Good: `storage.Reader`.

Reply:
Agreed, took this from the K8S doc. Updated it.